src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/items/ItemContainerContentsResourceHandler.java

```diff
@@ -0,0 +1,169 @@
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.resources.ResourceStack;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+
+/**
+ * Wraps the vanilla ComponentData of {@link ItemContainerContents} to allow it to be used as a {@link net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler IResourceHandler}
+ */
+// A very verbose name, but likely the most accurate
+public class ItemContainerContentsResourceHandler implements IResourceHandler<ItemResource> {
+    protected final int size;
+    protected final DataComponentType<ItemContainerContents> componentType;
+    protected final IItemContext itemContext;
+
+    public ItemContainerContentsResourceHandler(IItemContext itemContext, DataComponentType<ItemContainerContents> componentType, int size) {
+        this.componentType = componentType;
+        this.itemContext = itemContext;
+        this.size = size;
+    }
+
+    public ItemContainerContents getContents() {
+        return itemContext.getResource().getOrDefault(componentType, ItemContainerContents.fromItems(NonNullList.withSize(size(), ItemStack.EMPTY)));
+    }
+
+    public int setAndValidate(ItemContainerContents contents, int changedAmount, TransactionContext context) {
+        return itemContext.exchange(itemContext.getResource().with(componentType, contents), 1, context) == 1 ? changedAmount : 0;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public ItemResource getResource(int index) {
+        // You may need to still add what you had before. Depends on if this really resulted in much better
+        //return getContents().getImmutableStackInSlot(index).resource();
+        return ItemResource.of(getContents().getStackInSlot(index));
+    }
+
+    @Override
+    public int getAmount(int index) {
+        return getContents().getStackInSlot(index).getCount();
+    }
+
+    @Override
+    public int getCapacity(int index, ItemResource resource) {
+        Objects.checkIndex(index, size());
+        if (resource.isEmpty()) return Item.ABSOLUTE_MAX_STACK_SIZE;
+        return Math.min(resource.getMaxStackSize(), Item.ABSOLUTE_MAX_STACK_SIZE);
+    }
+
+    @Override
+    public boolean isValid(int index, ItemResource resource) {
+        ItemResource current = getResource(index);
+        return current.isEmpty() || current.equals(resource);
+    }
+
+    @Override
+    public boolean supportsInsertion(int index) {
+        return true;
+    }
+
+    @Override
+    public boolean supportsExtraction(int index) {
+        return true;
+    }
+
+    //TODO the resource should be in theory ignored here, but I want to be sure.
```

Comment:
What even uses this, and what is this method supposed to do? It is impossible to tell if the resource should be ignored, as it seems this entire method is ignored